src/plugins/flexibleLayout/components/frame.vue

```diff
@@ -21,104 +21,84 @@
-
-            <object-view
-                class="c-fl-frame__object-view"
-                :object="frame.domainObject">
-            </object-view>
+            <subobject-view
+                v-if="item.domainObject.identifier"
+                :item="item">
+            </subobject-view>
 
             <div class="c-fl-frame__size-indicator"
                  v-if="isEditing"
-                 v-show="frame.height && frame.height < 100">
-                {{frame.height}}%
+                 v-show="frame.size && frame.size < 100">
+                {{frame.size}}%
             </div>
         </div>
-
-        <drop-hint
-             v-show="isEditing && isDragging"
-             class="c-fl-frame__drop-hint"
-             :class="{'is-dragging': isDragging}"
-             @object-drop-to="dropHandler">
-        </drop-hint>
     </div>
 </template>
 
 <script>
-import ObjectView from '../../../ui/components/layout/ObjectView.vue';
-import DropHint from './dropHint.vue';
 import ResizeHandle from './resizeHandle.vue';
-import FrameHeader from '../../../ui/components/utils/frameHeader.vue';
+import SubobjectView from '../../displayLayout/components/SubobjectView.vue';
+import isEditingMixin from '../mixins/isEditing';
 
 export default {
     inject: ['openmct', 'domainObject'],
-    props: ['frame', 'index', 'containerIndex', 'isEditing', 'isDragging'],
+    props: ['frame', 'index', 'containerIndex'],
+    mixins: [isEditingMixin],
     data() {
         return {
+            item: {domainObject: {}},
             noFrame: this.frame.noFrame
         }
     },
     components: {
-        ObjectView,
-        DropHint,
         ResizeHandle,
-        FrameHeader
+        SubobjectView
     },
     methods: {
-        initDrag(event) {
-            this.$emit('frame-drag-from', this.index);
-        },
-        dropHandler(event) {
-            this.$emit('frame-drop-to', this.index, event);
+        setDomainObject(object) {
+            this.item.domainObject = object;
+            this.setSelection();
         },
-        continueDrag(event) {
-            if (!this.isDragging) {
-                this.isDragging = true;
-            }
+        setSelection() {
+           
+            let context = {
+                item: this.item.domainObject,
+                addContainer: this.addContainer,
+                type: 'frame',
+                index: this.index
+            };
+
+            this.unsubscribeSelection = this.openmct.selection.selectable(this.$refs.frame, context, false);
+            
+            this.openmct.objects.observe(this.domainObject, `configuration.containers[${this.containerIndex}].frames[${this.index}].noFrame`, this.toggleFrame);
+
         },
-        deleteFrame() {
-            this.$emit('delete-frame', this.index);
+        initDrag(event) {
+            this.$emit('frame-drag-from', this.index);
+            event.dataTransfer.setData('frameid', JSON.stringify([this.frame.id, this.containerIndex]));
```

Comment:
I think it's easier & clearer to call `event.dataTransfer.setData('frameid', this.frame.id);` and `event.dataTransfer.setData('containerIndex', this.containerIndex);` so that you don't have to do json parsing and stringifying.

Also, could you do this with `containerId` instead of `containerIndex`?  Less information to pass.

Reply:
I think its easier to locate the container with index instead of having to loop through the array, since we already have the index handy.

Reply:
Ahh, gotcha, I see it's also easier in the toolbar provider for generating mutation paths.